addons/product_images/wizard/product_get_pic_using_barcode.py

```diff
@@ -0,0 +1,219 @@
+                              "on %s", format_datetime(self.env, date)))
+
+        default = super().default_get(fields_list)
+
+        google_custom_search_cx = bool(
+            self.env['ir.config_parameter'].get_param('google.custom_search.cx')
+        )
+        google_custom_search_key = bool(
+            self.env['ir.config_parameter'].get_param('google.custom_search.key')
+        )
+        if not google_custom_search_cx or not google_custom_search_key:
+            raise UserError(_("You didn't configure properly your API Key or Search Engine ID."))
+
+        if self._context.get('active_model') == 'product.template':
+            product_ids = self.env['product.template'].browse(
+                self._context.get('active_ids')
+            ).product_variant_ids.filtered(lambda p: not p.image_1920 and p.barcode)
+        else:
+            product_ids = self.env['product.product'].browse(
+                self._context.get('active_ids')
+            ).filtered(lambda p: not p.image_1920 and p.barcode).ids
+        if not product_ids:
+            raise UserError(_("There isn't any product without a picture and with a barcode."))
+
+        nb_product = len(product_ids)
+        if nb_product > 10000:
+            product_ids = product_ids[:10000]
+
+        default['product_ids'] = product_ids
+        # Non computed because the value of product_ids doesn't yet exists.
+        default['nb_product'] = nb_product
+
+        return default
+
+    product_ids = fields.Many2many(
+        comodel_name='product.product',
+        help="List of the selected product without a picture and with a barcode",
+    )
+    nb_product = fields.Integer(string="Number of product selected", readonly=True)
+
+    def cron_get_pic_using_barcode(self):
+        """ Trigger a cron job to retrieve images on Google
+
+        Fetch the 10 first images to improve the user point of vue. This way, he can immediately
+        view the images and be assured that the task is running well. Also, if any error occurs,
+        they can be thrown to the user. Then, a cron job is trigger to be run as soon as possible,
+        unless the daily request limit is reach. In that case, the cron job will be schedule to run
+        a day later.
+
+        :raises UserError: If the project is misconfigure on Google's side
+        :raises UserError: If the API Key or Seach Engine ID is incorrect
+        :return: A notification to inform the user that the task is triggered
+        :rtype: `ir.actions.client`
+        """
+        self.product_ids.to_be_processed_for_image = True
+        try:
+            self._get_pics_from_google(self.product_ids[:10])
+        except HTTPError as http_error:
+            if http_error.response.status_code == requests.codes.too_many_requests:
+                self.env.ref('product_images.ir_cron_get_pics_from_google')._trigger(
+                    fields.datetime.now() + timedelta(days=1.0)
```

Comment:
```suggestion
                    fields.Datetime.now() + timedelta(days=1.0)
```
fields.datetime.now == datetime.now whereas fields.Datetime.now removes microseconds precision (which isn't needed here).